src/steamship/agents/schema/tool.py

```diff
@@ -47,3 +53,25 @@ def run(self, tool_input: List[Block], context: AgentContext) -> Union[List[Bloc
     def post_process(self, async_task: Task, context: AgentContext) -> List[Block]:
         """Transforms Task output into a List[Block]."""
         pass
+
+    def as_openai_function(self) -> OpenAIFunction:
```

Comment:
Minor nit: Does this belong in Tool, or in some OpenAI specific subclass of tool?

Reply:
My initial reaction is that all Tools should be able to be translated. This is really just a schema thing. It _feels_ like having to go back and decide which tools to make OpenAI tools introduces cognitive (and other) overhead.

I could see having this be on the OpenAIFunction class, maybe. Like, `OpenAIFunction.from_tool(Tool)` maybe.

The nice thing about having it be on Tool however is that individual Tools could decide to override the generation in a tailored fashion.

The other bit I've been thinking about is whether or not this should return a list of OpenAIFunctions, so that a tool could potentially represent a range of functions. For the moment, I figure that can be handled with an enum parameter though.

Reply:
+1 to make this available for all tools

Reply:
Fine by me :)